### PR TITLE
#7152 Replace instance of fetch for `/api/bricks` with RTK Query Hook

### LINKED ISSUE #7152: Refactor non-RTK-Query fetch calls to use RTK Query
## User Story

As a PixieBrix engineer, I want as many fetch calls as possible to use RTK Query, so that I have a standardized method for making requests. 

## Motivation

We want a single abstraction for fetching, adapting, and caching application data from PB APIs. See https://www.notion.so/pixiebrix/Extension-API-Service-Layer-9e55ef97d7bd4d20a48f5f3ae8c23c32

## Endpoints
- https://www.notion.so/pixiebrix/Extension-API-Service-Layer-9e55ef97d7bd4d20a48f5f3ae8c23c32?pvs=4#d0619bbe816a42db970bee69525f4b86

### PR DESCRIPTION
**Pull request by mnholtz**
## What does this PR do?

- Closes #7152
- It seems like this was the only place we use `fetch` in React code?

## Demo

- https://www.loom.com/share/45310d7368d74bb2b47cdd7f92eae554

## Checklist

- [x] Add tests
- [x] Designate a primary reviewer @grahamlangford

### CODE REVIEW

File: src/extensionConsole/pages/brickEditor/Editor.tsx [hunk 1]
```diff
@@ -73,18 +72,37 @@ interface OwnProps {
...
+  const [getEditablePackages] =
+    appApi.endpoints.getEditablePackages.useLazyQuery();
+
+  return useCallback(
+    async (brickId: RegistryId) => {
+      let brick;
+
+      try {
+        const editablePackages = await getEditablePackages(
+          undefined,
```
Comment (mnholtz) [inline src/extensionConsole/pages/brickEditor/Editor.tsx:85]: Why `undefined`? See https://pixiebrix.slack.com/archives/C023KL47XV4/p1677610197731169

I don't think we ever made an eslint rule for this. Thoughts @grahamlangford?

Comment (grahamlangford) [reply src/extensionConsole/pages/brickEditor/Editor.tsx:85]: This is a great callout. I'd love to get a linting rule in for this, but I'm not sure how we'd write it for this exact case. Writing something that matches on `use\S+Query` as mentioned in the slack thread makes a lot of sense, but I'm not sure how we'd write it to include `getEditablePackages`. Do you have any ideas?

**Review by grahamlangford — APPROVED** [review]

File: src/extensionConsole/pages/brickEditor/Editor.tsx (hunk 1 above)
Comment (mnholtz) [reply src/extensionConsole/pages/brickEditor/Editor.tsx:85]: Good point. I'm not sure, as I don't have experience writing lint rules. I'll make sure to outline this desire on the ticket.

Comment (mnholtz) [reply src/extensionConsole/pages/brickEditor/Editor.tsx:85]: Ticket is here: https://github.com/pixiebrix/pixiebrix-extension/issues/7170

**Pull request merged**